odxtools/element.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from typing import List, Optional, TypedDict, cast
+from xml.etree import ElementTree
+
+from odxtools.exceptions import odxrequire
+from odxtools.utils import create_description_from_et
+
+from .odxlink import OdxDocFragment, OdxLinkId
+
+
+class BaseElementKwargs(TypedDict):
+    short_name: str
+    long_name: Optional[str]
+    description: Optional[str]
+
+
+@dataclass
+class NamedElement:
+    short_name: str
+    long_name: Optional[str]
+    description: Optional[str]
```

Comment:
the TypedDict and the dataclass exhibit essentially the same attributes. it would be great if that repetition  could somehow be avoided. (if this is impossible, so be it...)

Reply:
okay, it can be done using only dataclasses and mypy also seems to get it correctly:
```python
from dataclasses import dataclass, asdict

@dataclass
class Foo:
    hello: int

@dataclass
class Bar(Foo):
    world: float

print(Bar(**asdict(Foo(1)), world=2.0))
# Bar(hello=1, world=2.0)
```

(I guess it will not work if the base class is abstract and cannot be instantiated, though.)

Reply:
are you sure mypy is not just ignoring it?

Reply:
it works:
```python
from dataclasses import dataclass, asdict

@dataclass
class Foo:
    hello: int

@dataclass
class Bar(Foo):
    world: float

print(Bar(**asdict(Foo(1)), world=2.0)) # ok
print(Bar(**asdict(Foo("foobar")), world=2.0)) # wrong type for "hello"
print(Bar(world=2.0)) # missing "hello"
```

produces
```
> mypy bla.py 
bla.py:12: error: Argument 1 to "Foo" has incompatible type "str"; expected "int"
bla.py:13: error: Missing positional argument "hello" in call to "Bar"
Found 2 errors in 1 file (checked 1 source file)
```

(sometimes I think that `mypy` is close to passing the turing test. usually that feeling goes away a minute later when it complains about something stupid ;))

Reply:
asdict is recursive, it will convert sub properties to dict as well

Reply:
maybe replacing `asdict()` with

```python
def extract_attribs(obj: Any) -> Dict[str, Any]:
    return { x.name: getattr(obj, x.name) for x in obj.__dataclass_fields__.values() }
```

it still seems to work with mypy...